plugins/scala_test/recipes.py

```diff
@@ -0,0 +1,83 @@
+from polyglot_piranha import Rule, OutgoingEdges, RuleGraph, PiranhaArguments, execute_piranha
+
+def replace_imports(
+    target_new_types: dict[str, str], search_heuristic: str, path_to_codebase: str,
+    dry_run = False
+):
+    find_relevant_files = Rule(
+        name="find_relevant_files",
+        query="((identifier) @x (#eq? @x \"@search_heuristic\"))",
+        holes={"search_heuristic"},
+    )
+    e1 = OutgoingEdges("find_relevant_files", to=[f"update_import"], scope="File")
+
+    rules = [find_relevant_files]
+    edges = [e1]
+
+    for target_type, new_type in target_new_types.items():
+        rs, es = replace_import_rules_edges(target_type, new_type)
+        rules.extend(rs)
+        edges.extend(es)
+
+    rule_graph = RuleGraph(rules=rules, edges=edges)
+
+    args= PiranhaArguments(
+        language="scala",
+        path_to_codebase=path_to_codebase,
+        rule_graph=rule_graph,
+        substitutions={"search_heuristic": f"{search_heuristic}"},
+        dry_run=dry_run
+    )
+    
+    return execute_piranha(args)
+    
+
+
+def replace_import_rules_edges(
```

Comment:
```suggestion
def replace_import_rules_and_edges(
```

Otherwise it reads as generating the "rules' edges" and it might be surprising that it also returns a list of rules.

Btw, these methods and `update_imports` could use docs. Maybe even those in `test_update_imports` too, but I leave that up to you.

Reply:
done! Credits - CoPilot